### PR TITLE
feat: wire-up zapper read-only staking opportunities

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

- [x] Make Yearn great again - ish, she's great again but we're not doing LP for this PoC, to be added as a direct follow-up
- [x] Flag
- [x] Support all our supported EVM networks
- [x] Missing balances for read-only opportunities
- [x] Hardcoded underlyingAssetRatios - do we need this? We're not going to open a modal with underlyingAssetIds, but may still need this for equity to work
- [x] Remove staking hardcoding i.e detect LP/staking
- [x] Read-only opportunities click links to URL - apps meta to be fetched from `https://api.zapper.xyz/v2/apps`
- [x] Handle holistically, not only in Earn by provider - possibly
- [x] Better home for read-only opportunities fetching, related to the previous point
- [x] Sorting  - needs to be merged to current non-read-only opportunities
- [x] Dupes - filter out existing opportunities for now. As a follow-up, we will want to do one better and tweak the concept of "read-only". We are effectively double-fetching UNI-V2 (we could only fetch from the "read-only" resolvers and `.select()` the cached query data in UNI-V2 resolvers) and Idle (this one is unfortunate but we will still want to double-fetch here, since `investor-idle` does a bit more and fetches from Idle API, which contains more data)
- [ ] ~~Handle rewards/claimable~~ Not yet, tbd in a direct follow-up stack

This PR brings base support (crypto/fiat/underlying asset amounts and minimal assets upsertion) for read-only staking opportunities from Zapper.

While at it, it does a bunch of housekeeping to circumvent circular dependencies issues, and refactors previous concepts of the opportunitiesSlice, namely:

- A `provider` is now a string for all intents and purposes, and most consumptions of `DefiProvider` are now removed. `DefiProvider` will eventually be LSP renamed to `KnownDefiProvider`, avoided in this PR to avoid making it larger
- A `StakingId` can now be an AssetId nominal, or it can be an AssetId nominal, serialized with a discriminating key:
https://github.com/shapeshift/web/blob/b4353a7e534fb6bca26867e95b0cd89c541fb5dc/src/state/slices/opportunitiesSlice/types.ts#L104-L108
The same concept will be brought into LP when implemented as a follow-up. The rationale being this is opportunities can share the same AssetId. This was not a problem previously as we were lucky enough to have unique `AssetId`ish for all opportunities, but take a look at EPNS for example:

```json
    {
        "key": "750724700",
        "address": "0x05a1ff0a32bc24265bcb39499d0c5d9a6cb2011c",
        "appId": "epns",
        "appName": "EPNS",
        "appImage": "https://storage.googleapis.com/zapper-fi-assets/apps/epns.png",
        "network": "ethereum",
        "updatedAt": "2023-05-22T13:10:01.793Z",
        "balanceUSD": 2353.881018669655,
        "products": [
            {
                "label": "Farms",
                "assets": [
                    {
                        "key": "3053580903",
                        "type": "contract-position",
                        "appId": "epns",
                        "groupId": "farm",
                        "network": "ethereum",
                        "address": "0xb72ff1e675117bedeff05a7d0a472c3844cfec85",
                        "tokens": [
                            {
                                "metaType": "supplied",
                                "type": "app-token",
                                "appId": "uniswap-v2",
                                "price": 60.20732497712855,
                                "supply": 22783.534617638885,
                                "symbol": "UNI-V2",
                                "tokens": [
                                    {
                                        "type": "base-token",
                                        "price": 1817.28,
                                        "symbol": "WETH",
                                        "address": "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2",
                                        "network": "ethereum",
                                        "decimals": 18,
                                        "balance": 0.4941023496788433,
                                        "balanceRaw": "494102349678843300",
                                        "balanceUSD": 897.9223180243683
                                    },
                                    {
                                        "type": "base-token",
                                        "price": 0.298894,
                                        "symbol": "PUSH",
                                        "address": "0xf418588522d5dd018b425e472991e52ebbeeeeee",
                                        "network": "ethereum",
                                        "decimals": 18,
                                        "balance": 3038.861373227592,
                                        "balanceRaw": "3038861373227592300000",
                                        "balanceUSD": 908.2974312894879
                                    }
                                ],
                                "address": "0xaf31fd9c3b0350424bf96e551d2d1264d8466205",
                                "groupId": "pool",
                                "network": "ethereum",
                                "decimals": 18,
                                "dataProps": {
                                    "apy": 0,
                                    "fee": 0.3,
                                    "isDebt": false,
                                    "volume": 0,
                                    "reserves": [
                                        375.24659961882134,
                                        2307866.7765045497
                                    ],
                                    "liquidity": 1371735.6728518426
                                },
                                "displayProps": {
                                    "label": "WETH / PUSH",
                                    "images": [
                                        "https://storage.googleapis.com/zapper-fi-assets/tokens/ethereum/0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2.png",
                                        "https://storage.googleapis.com/zapper-fi-assets/tokens/ethereum/0xf418588522d5dd018b425e472991e52ebbeeeeee.png"
                                    ],
                                    "statsItems": [
                                        {
                                            "label": "Fee",
                                            "value": {
                                                "type": "pct",
                                                "value": 0.3
                                            }
                                        },
                                        {
                                            "label": "Liquidity",
                                            "value": {
                                                "type": "dollar",
                                                "value": 1371735.6728518426
                                            }
                                        },
                                        {
                                            "label": "Reserves",
                                            "value": "375.25 / 2307866.78"
                                        },
                                        {
                                            "label": "Ratio",
                                            "value": "50% / 50%"
                                        },
                                        {
                                            "label": "Volume",
                                            "value": {
                                                "type": "dollar",
                                                "value": 0
                                            }
                                        },
                                        {
                                            "label": "APY",
                                            "value": {
                                                "type": "pct",
                                                "value": 0
                                            }
                                        },
                                        {
                                            "label": "Share",
                                            "value": {
                                                "type": "pct",
                                                "value": 0.13167403786756673
                                            }
                                        }
                                    ],
                                    "secondaryLabel": "50% / 50%"
                                },
                                "pricePerShare": [
                                    0.01647007832262811,
                                    101.29537910758641
                                ],
                                "key": "988708634",
                                "balance": 30,
                                "balanceRaw": "30000000000000000000",
                                "balanceUSD": 1806.2197493138565
                            },
                            {
                                "metaType": "claimable",
                                "type": "base-token",
                                "network": "ethereum",
                                "address": "0xf418588522d5dd018b425e472991e52ebbeeeeee",
                                "symbol": "PUSH",
                                "decimals": 18,
                                "price": 0.299324,
                                "balance": 292.3600489153441,
                                "balanceRaw": "292360048915344092289",
                                "balanceUSD": 87.51037928153646
                            }
                        ],
                        "dataProps": {
                            "liquidity": 766192.3594212142,
                            "apy": 0,
                            "isActive": true
                        },
                        "displayProps": {
                            "label": "WETH / PUSH",
                            "tertiaryLabel": "0.000% APY",
                            "images": [
                                "https://storage.googleapis.com/zapper-fi-assets/tokens/ethereum/0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2.png",
                                "https://storage.googleapis.com/zapper-fi-assets/tokens/ethereum/0xf418588522d5dd018b425e472991e52ebbeeeeee.png"
                            ],
                            "statsItems": [
                                {
                                    "label": "Liquidity",
                                    "value": {
                                        "type": "dollar",
                                        "value": 766192.3594212142
                                    }
                                }
                            ]
                        },
                        "balanceUSD": 1893.730128595393
                    },
                    {
                        "key": "2164279715",
                        "type": "contract-position",
                        "appId": "epns",
                        "groupId": "farm",
                        "network": "ethereum",
                        "address": "0xb72ff1e675117bedeff05a7d0a472c3844cfec85",
                        "tokens": [
                            {
                                "metaType": "supplied",
                                "type": "base-token",
                                "network": "ethereum",
                                "address": "0xf418588522d5dd018b425e472991e52ebbeeeeee",
                                "symbol": "PUSH",
                                "decimals": 18,
                                "price": 0.299324,
                                "balance": 863,
                                "balanceRaw": "863000000000000000000",
                                "balanceUSD": 258.31661199999996
                            },
                            {
                                "metaType": "claimable",
                                "type": "base-token",
                                "network": "ethereum",
                                "address": "0xf418588522d5dd018b425e472991e52ebbeeeeee",
                                "symbol": "PUSH",
                                "decimals": 18,
                                "price": 0.299324,
                                "balance": 674.3003503703754,
                                "balanceRaw": "674300350370375396550",
                                "balanceUSD": 201.83427807426224
                            }
                        ],
                        "dataProps": {
                            "liquidity": 1993313.4504295196,
                            "apy": 0,
                            "isActive": true
                        },
                        "displayProps": {
                            "label": "PUSH",
                            "tertiaryLabel": "0.000% APY",
                            "images": [
                                "https://storage.googleapis.com/zapper-fi-assets/tokens/ethereum/0xf418588522d5dd018b425e472991e52ebbeeeeee.png"
                            ],
                            "statsItems": [
                                {
                                    "label": "Liquidity",
                                    "value": {
                                        "type": "dollar",
                                        "value": 1993313.4504295196
                                    }
                                }
                            ]
                        },
                        "balanceUSD": 460.1508900742622
                    }
                ],
                "meta": []
            }
        ]
    }
```

Note how both EPNS PUSH staking and WETH / PUSH share the same address, thus using the same `StakingId` for both would result in us being able to display only one.
## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A, see https://www.notion.so/Displaying-Earn-DeFi-positions-5fbfca607cd3491cb621e87242328209

## Risk

- UniV2 and/or current opportunities might be broken, test accordingly with the flag on and off

## Testing

- Test both with the flag on and off, there should be no user-facing diff with the flag off

- With a wallet that's *staked* into many farming opportunities, ensure these opportunities are fetched and the data is the same as in Zapper
- Opportunities meta/data is correct i.e staked balance, name, APY, provider name, icon and link
- Clicking the opportunity in "By Provider", or "Manage" in "By Asset' links you to the opportunity's dApp in a new tab
- Opportunities are still sorted properly i.e by balance, then APY for the inactive
- Current opportunities (i.e Idle, UNI-V2) still open their modal, *not* the external link for them
- There are no duplicate opportunities with the current ones (i.e UNI-V2, Idle)

## Screenshots (if applicable)

image
image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4491** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4491" target="_blank">Graphite</a> 
    * **PR #4504** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4504" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4504?utm_source=stack-comment).

File: src/components/StakingVaults/WalletProviderTable.tsx [hunk 1]
```diff
@@ -57,6 +62,7 @@ export const WalletProviderTable: React.FC<ProviderTableProps> = ({
     return (
       <Flex gap={4} flexDir='column'>
         {rows.map((row, index) => (
+          // @ts-ignore TODO(gomes): fixme
```
Comment (gomesalexandre) [inline src/components/StakingVaults/WalletProviderTable.tsx:65]: Actually fix me

File: src/state/slices/opportunitiesSlice/selectors/readonly.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        ).concat(item.opportunityId),
+      }
+
+      return {
+        ...acc,
+        [provider]: {
+          provider,
+          apy,
+          fiatAmount: totalFiatAmount.toString(),
+          // @ts-ignore TODO(gomes): implement fiatRewardsAmount
```
Comment (gomesalexandre) [inline src/state/slices/opportunitiesSlice/selectors/readonly.ts:85]: ts-ignore -> TODO

File: src/state/slices/opportunitiesSlice/selectors/readonly.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+          fiatRewardsAmount: bnOrZero(item.fiatRewardsAmount).toString(),
+          // @ts-ignore TODO(gomes): implement fiatRewardsAmount
```
Comment (gomesalexandre) [inline src/state/slices/opportunitiesSlice/selectors/readonly.ts:87]: ditto

File: src/components/EarnDashboard/components/ProviderDetails/OpportunityRow.tsx [hunk 4]
```diff
@@ -98,6 +99,11 @@ export const OpportunityRow: React.FC<
 
   const handleClick = useCallback(
     (action: DefiAction) => {
+      if (opportunity.isReadOnly) {
+        const url = getMetadataForProvider(opportunity.provider)?.url
+        url && window.open(url, '_blank')
```
Comment (gomesalexandre) [inline src/components/EarnDashboard/components/ProviderDetails/OpportunityRow.tsx:104]: should be implemented in "By Asset" as well

**Comment by gomesalexandre** [comment]
Opening for early review

**Comment by gomesalexandre** [comment]
Opening for early review

File: src/components/EarnDashboard/components/ProviderDetails/WalletLpByAsset.tsx [hunk 5]
```diff
@@ -34,6 +35,9 @@ export const WalletLpByAsset: React.FC<WalletLpByAssetProps> = ({ ids }) => {
   } = useWallet()
   const assets = useAppSelector(selectAssets)
   const lpOpportunities = useAppSelector(selectAggregatedEarnUserLpOpportunities)
+
+  useGetZapperAppsBalancesOutputQuery()
```
Comment (0xdef1cafe) [inline src/components/EarnDashboard/components/ProviderDetails/WalletLpByAsset.tsx:39]: why not in `AppProvider`?

Comment (0xdef1cafe) [reply src/components/EarnDashboard/components/ProviderDetails/WalletLpByAsset.tsx:39]: e.g. navigate to dashboard, then asset page, this won't be populated. opportunities data is used in way more than just this view

File: src/components/EarnDashboard/components/ProviderDetails/WalletStakingByAsset.tsx [hunk 6]
```diff
@@ -42,7 +43,11 @@ export const WalletStakingByAsset: React.FC<StakingPositionsByAssetProps> = ({ i
   const stakingOpportunities = useAppSelector(
     selectAggregatedEarnUserStakingOpportunitiesIncludeEmpty,
   )
-  const filteredDown = stakingOpportunities.filter(e => ids.includes(e.id as OpportunityId))
+
+  useGetZapperAppsBalancesOutputQuery()
+  const filteredDown = stakingOpportunities
+    // @ts-ignore FIXME
```
Comment (0xdef1cafe) [inline src/components/EarnDashboard/components/ProviderDetails/WalletStakingByAsset.tsx:49]: are we fixable?

File: src/components/StakingVaults/WalletProviderTable.tsx [hunk 7]
```diff
@@ -40,6 +41,10 @@ export const WalletProviderTable: React.FC<ProviderTableProps> = ({
     selectAggregatedEarnOpportunitiesByProvider(state, rowsFilter),
   )
 
+  // Only for fetching - we're consumed derived data once fetched and cached
+  // Maybe find a better home for this?
```
[the remark this answers is not in the document]
Comment (0xdef1cafe) [reply src/components/StakingVaults/WalletProviderTable.tsx:45]: yessir

File: src/state/slices/opportunitiesSlice/selectors/combined.ts [hunk 8]
```diff
@@ -306,8 +306,8 @@ export const selectAggregatedEarnOpportunitiesByProvider = createDeepEqualOutput
     searchQuery,
   ): AggregatedOpportunitiesByProviderReturn[] => {
     if (isEmpty(marketData)) return []
-    const totalFiatAmountByProvider = {} as Record<DefiProvider, BN>
-    const projectedAnnualizedYieldByProvider = {} as Record<DefiProvider, BN>
+    const totalFiatAmountByProvider = {} as Record<string, BN>
```
Comment (0xdef1cafe) [inline src/state/slices/opportunitiesSlice/selectors/combined.ts:309]: assing over cast

File: src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts [hunk 9]
```diff
@@ -63,7 +64,8 @@ export const selectStakingIds = createDeepEqualOutputSelector(
 export const selectUserStakingIds = createDeepEqualOutputSelector(
   selectWalletAccountIds,
   (state: ReduxState) => state.opportunities.userStaking.ids,
-  (walletAccountIds, userStakingIds): UserStakingId[] =>
+  selectGetReadOnlyOpportunities,
+  (walletAccountIds, userStakingIds, _readOnlyOpportunitays): UserStakingId[] =>
```
Comment (0xdef1cafe) [inline src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts:68]: remove

File: src/state/slices/opportunitiesSlice/types.ts [hunk 10]
```diff
@@ -78,14 +101,18 @@ export type UserStakingOpportunity =
 export type UserStakingOpportunityWithMetadata = UserStakingOpportunity & OpportunityMetadata
 
 // The AccountId of the staking contract in the form of chainId:accountAddress
-export type StakingId = Nominal<string, 'StakingId'> & AssetId
+export type StakingIdBase = Nominal<string, 'StakingId'> & AssetId
+// Sometimes, an AssetIdish (i.e a a CAIP-19) is not enough to uniquely identify a staking opportunity
+// So we add a StakingKey to the StakingId to make it unique
+export type StakingKey = Nominal<string, 'StakingKey'> & string
```
Comment (0xdef1cafe) [inline src/state/slices/opportunitiesSlice/types.ts:107]:
```suggestion
export type StakingKey = Nominal<string, 'StakingKey'>
```

previous `& AssetId` pattern was just such that `StakingId` becomes a superset of `AssetId`

File: src/state/slices/opportunitiesSlice/types.ts [hunk 11]
```diff
@@ -78,14 +101,18 @@ export type UserStakingOpportunity =
...
+export type StakingId = StakingIdBase | `${StakingIdBase}#${StakingKey}`
 // The AccountId of the LP contract in the form of chainId:accountAddress
 export type LpId = Nominal<string, 'LpId'> & AssetId
 
 export type ValidatorId = Nominal<string, 'ValidatorId'> & AccountId
 export type OpportunityId = LpId | StakingId | ValidatorId
 // The unique identifier of an lp opportunity in the form of UserAccountId*StakingId
-export type UserStakingId = `${AccountId}*${StakingId}`
+export type UserStakingId = `${AccountId}*${StakingId}` | `${AccountId}*${StakingId}`
```
[the remark this answers is not in the document]
Comment (0xdef1cafe) [reply src/state/slices/opportunitiesSlice/types.ts:115]: the ole identity union

File: src/components/EarnDashboard/components/ProviderDetails/WalletStakingByAsset.tsx (hunk 6 above)
Comment (gomesalexandre) [reply src/components/EarnDashboard/components/ProviderDetails/WalletStakingByAsset.tsx]: d1ad870ca301e83c11b8ba0ed54110be2fbcc26e

File: src/state/slices/opportunitiesSlice/selectors/combined.ts (hunk 8 above)
Comment (gomesalexandre) [reply src/state/slices/opportunitiesSlice/selectors/combined.ts]: d1ad870ca301e83c11b8ba0ed54110be2fbcc26e

File: src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts (hunk 9 above)
Comment (gomesalexandre) [reply src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts]: d1ad870ca301e83c11b8ba0ed54110be2fbcc26e

File: src/state/apis/zapper/zapperApi.ts [hunk 12]
```diff
@@ -240,7 +303,298 @@ export const zapper = createApi({
...
+              const accountId = toAccountId({
+                chainId:
+                  ZAPPER_NETWORKS_TO_CHAIN_ID_MAP[
+                    appAccountBalance.network as SupportedZapperNetwork
+                  ],
+                account: appAccountBalance.address,
+              })
+
+              // [0] only for debugging, obviously support all products eventually
+              const appAccountOpportunities = (appAccountBalance.products[0]?.assets ?? [])
```
Comment (0xdef1cafe) [inline src/state/apis/zapper/zapperApi.ts:356]: use all now? is this comment invalid?

File: src/state/slices/opportunitiesSlice/types.ts (hunk 10 above)
Comment (gomesalexandre) [reply src/state/slices/opportunitiesSlice/types.ts]: e05b6759b

File: src/state/apis/zapper/zapperApi.ts [hunk 13]
```diff
@@ -240,7 +303,298 @@ export const zapper = createApi({
...
+                      const underlyingAsset = makeAsset({
+                        assetId: underlyingAssetId,
+                        symbol: asset.tokens[i].symbol,
+                        // No dice here, there's no name property
+                        name: asset.tokens[i].symbol,
+                        precision: asset.tokens[i].decimals,
+                        icon: asset.displayProps?.images[i] ?? '',
+                      })
+
+                      // TODO(gomes): we might want to do this in a batch
```
Comment (0xdef1cafe) [inline src/state/apis/zapper/zapperApi.ts:415]: let's do this now, upserting to assets will cause every selector to recompute

File: src/state/apis/zapper/zapperApi.ts [hunk 14]
```diff
@@ -240,6 +303,296 @@ export const zapper = createApi({
...
+
+          // Populate read only metadata payload
+          for (const id in readOnlyMetadata) {
+            if (readOnlyMetadata[id].type === 'staking') {
+              // TODO(gomes): lp too
+              metadataUpsertPayload.byId[id] = readOnlyMetadata[id]
+            }
+          }
+
+          // Populate read only userData payload
```
Comment (0xdef1cafe) [inline src/state/apis/zapper/zapperApi.ts:553]: mmmm smells like LLM and not snails in here

File: src/context/AppProvider/AppContext.tsx [hunk 15]
```diff
@@ -192,6 +192,8 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
           case osmosisChainId:
           case avalancheChainId:
             ;(async () => {
+              dispatch(zapper.endpoints.getZapperAppsBalancesOutput.initiate())
```
Comment (0xdef1cafe) [inline src/context/AppProvider/AppContext.tsx:195]: we still can't move this up to 183?

Comment (gomesalexandre) [reply src/context/AppProvider/AppContext.tsx]: https://github.com/shapeshift/web/pull/4504/commits/6e9a2e096bec2b52fdea6aba12b7958653b71007 absolootely can

File: src/context/AppProvider/AppContext.tsx [hunk 16]
```diff
@@ -192,6 +192,8 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
           case osmosisChainId:
           case avalancheChainId:
             ;(async () => {
+              zapperApi.endpoints.getZapperAppsOutput.initiate()
```
[the remark this answers is not in the document]
Comment (gomesalexandre) [reply src/context/AppProvider/AppContext.tsx]: https://github.com/shapeshift/web/pull/4504/commits/6e9a2e096bec2b52fdea6aba12b7958653b71007

File: src/state/apis/zapper/zapperApi.ts (hunk 13 above)
Comment (gomesalexandre) [reply src/state/apis/zapper/zapperApi.ts]: https://github.com/shapeshift/web/pull/4504/commits/6e9a2e096bec2b52fdea6aba12b7958653b71007

File: src/state/apis/zapper/zapperApi.ts [hunk 17]
```diff
@@ -240,7 +303,298 @@ export const zapper = createApi({
...
+                  const apy = bnOrZero(asset.dataProps?.apy).div(100).toString()
+                  const tvl = bnOrZero(asset.dataProps?.liquidity).toString()
+                  const icon = asset.displayProps?.images?.[0] ?? ''
+                  const name = asset.displayProps?.label ?? ''
+                  const groupId = asset.groupId
+                  const type = asset.type
+                  const assetId = toAssetId({
+                    chainId:
+                      ZAPPER_NETWORKS_TO_CHAIN_ID_MAP[asset.network as SupportedZapperNetwork],
+                    assetNamespace: 'erc20', // TODO(gomes): this might break for native assets, is that a thing?,
```
[the remark this answers is not in the document]
Comment (gomesalexandre) [reply src/state/apis/zapper/zapperApi.ts]: https://github.com/shapeshift/web/pull/4504/commits/6e9a2e096bec2b52fdea6aba12b7958653b71007

File: src/state/apis/zapper/zapperApi.ts (hunk 12 above)
Comment (gomesalexandre) [reply src/state/apis/zapper/zapperApi.ts]: https://github.com/shapeshift/web/pull/4504/commits/6e9a2e096bec2b52fdea6aba12b7958653b71007

File: src/state/slices/opportunitiesSlice/types.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (gomesalexandre) [reply src/state/slices/opportunitiesSlice/types.ts]: 28a43dfe3cf8fb8a008b9fca6ebde550419acd7b

File: .env.dev [hunk 18]
```diff
@@ -49,3 +49,7 @@ REACT_APP_MIDGARD_URL=https://dev-indexer.thorchain.shapeshift.com/v2
 
 # Covalent
 REACT_APP_FEATURE_COVALENT_JAYPEGS=true
+>>>>>>> origin/develop
```
Comment (0xdef1cafe) [inline .env.dev:52]: merp

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**